### PR TITLE
fix : Tabs `onEdit` Callback parameter type problem

### LINKED ISSUE #39925: [docs] :  Tabs: `onEdit`  callback type is wrong
### Reproduction link

[Edit on CodeSandbox](https://codesandbox.io/s/antd-reproduction-template-forked-lz9exn?file=/index.tsx)

### Steps to reproduce

The ts type error occurred in using the Tabs deletion function.

### What is expected?

Obtain the correct type under ts

### What is actually happening?

The type of the first parameter under ts gets an error warning

| Environment | Info |
| --- | --- |
| antd | 5.1.1 |
| React | 18.2.0 |
| System | win11 |
| Browser | Chrome  108.0.5359.125 |

---
I didn't see this problem in the codesandbox, but it only occurred locally.
image

image

### PR DESCRIPTION
**Pull request by RSS1102**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link
fix https://github.com/ant-design/ant-design/issues/39925

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |           |
| 🇨🇳 Chinese |           |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [ ] Doc is updated/provided or not needed
- [ ] Demo is updated/provided or not needed
- [ ] TypeScript definition is updated/provided or not needed
- [ ] Changelog is provided or not needed

### CODE REVIEW

File: components/tabs/demo/custom-add-trigger.tsx [hunk 1]
```diff
@@ -21,7 +21,7 @@ const App: React.FC = () => {
     setActiveKey(newActiveKey);
   };
 
-  const remove = (targetKey: string) => {
+  const remove = (targetKey: React.MouseEvent | React.KeyboardEvent | string) => {
```
Comment (yoyo837) [inline components/tabs/demo/custom-add-trigger.tsx:24]: 提取一下 type呗

**Comment by afc163** [comment]
type.test.tsx 里加个 type 测试

File: components/tabs/demo/editable-card.tsx [hunk 2]
```diff
@@ -49,7 +51,10 @@ const App: React.FC = () => {
     setActiveKey(newActiveKey);
   };
 
-  const onEdit = (targetKey: string, action: 'add' | 'remove') => {
+  const onEdit = (
+    targetKey: React.MouseEvent | React.KeyboardEvent | string,
```
Comment (yoyo837) [inline components/tabs/demo/editable-card.tsx:55]: 这里不能使用同一个提取的类型吗

Comment (RSS1102) [reply components/tabs/demo/editable-card.tsx:55]: 但是有单独使用` targetKey`的地方，我觉得两个类型来表示更好吧？

>

**Comment by RSS1102** [comment]
> type.test.tsx 里加个 type 测试

😕阿这...没写过jest....

File: components/tabs/demo/editable-card.tsx (hunk 2 above)
Comment (yoyo837) [reply components/tabs/demo/editable-card.tsx:55]: 只需要改demo吗?组件里该类型是否也需要提取?

Comment (RSS1102) [reply components/tabs/demo/editable-card.tsx:55]: > 组件内我倒是没有注意，我等会看下。

**Comment by afc163** [comment]
https://github.com/ant-design/ant-design/issues/39925#issuecomment-1375042294

**Review by afc163 — APPROVED** [review]

**Comment by RSS1102** [comment]
啊这....不是..
@yoyo837让我把所有的类型进行一个 提取的类型。我只做了docs的，对于代码的类型我还l没来得及做（最近公司工作忙）。
 @afc163是不是应再等等再合并呢？

**Comment by afc163** [comment]
可以再发个 PR 来处理。

**Comment by RSS1102** [comment]
> 可以再发个 PR 来处理。

get

**Pull request merged**